### ISSUE TITLE
ensure_member_controller_account_origin mock, Incomplete implementation 

### ISSUE THREAD

**Issue by L3pereira** [Joystream/joystream#3325]
@shamil-gadelshin 
I found that the function implementation of ```ensure_member_controller_account_origin``` used in tests, is not correct and is not reflecting the real outputs that this function has.
https://github.com/Joystream/joystream/blob/1c37a713511cfaf35b9577069f78827492324f2c/runtime-modules/bounty/src/tests/mocks.rs#L209-L222

In reality we validate the origin eg. in the ```cancel_bounty(origin, creator: BountyActor<MemberId<T>>, bounty_id: T::BountyId)``` with 
https://github.com/Joystream/joystream/blob/1c37a713511cfaf35b9577069f78827492324f2c/runtime-modules/bounty/src/lib.rs#L777-L780
Inside ```ensure_bounty_actor_manager``` we have the Membership implementation 
https://github.com/Joystream/joystream/blob/1c37a713511cfaf35b9577069f78827492324f2c/runtime-modules/bounty/src/actors.rs#L35

If we analyze what the above function does, we can see we will return UnsignedOrigin in case the origin is not a signed account. 
Then we validate the existence of an actor_id and his controller account (bellow), if the validation fails we return MemberProfileNotFound or ControllerAccountRequired.
https://github.com/Joystream/joystream/blob/1c37a713511cfaf35b9577069f78827492324f2c/runtime-modules/membership/src/lib.rs#L1102-L1130

This means that in tests the following function returns only Ok or BadOrigin error, but in reality the code will return one of the following errors 
ControllerAccountRequired
UnsignedOrigin
MemberProfileNotFound
https://github.com/Joystream/joystream/blob/1c37a713511cfaf35b9577069f78827492324f2c/runtime-modules/bounty/src/tests/mocks.rs#L210 

The test below I think is not correct because, in reality, the error should be ControllerAccountRequired or MemberProfileNotFound.
a ```NotBountyActor``` means that the ```ensure_bounty_actor_manager``` returned Ok(()) which is not true 
https://github.com/Joystream/joystream/blob/1c37a713511cfaf35b9577069f78827492324f2c/runtime-modules/bounty/src/tests/mod.rs#L611-L624

I propose the following implementation for tests
```rust
    fn ensure_member_controller_account_origin(
        origin: Origin,
        member_id: u64,
    ) -> Result<u128, DispatchError> {
        
         let signed_account_id = frame_system::ensure_signed(origin)?;

         if member_id >= 10 {
             return Err(membership::Error::<Test>::MemberProfileNotFound.into());
         }

         if (member_id as u128) != signed_account_id{
             return Err(membership::Error::<Test>::ControllerAccountRequired.into());    
        }

        return Ok(signed_account_id);


    }
```
```if member_id >= 10 ``` is because there is already a way to check if a membership id exits

https://github.com/Joystream/joystream/blob/1c37a713511cfaf35b9577069f78827492324f2c/runtime-modules/bounty/src/tests/mocks.rs#L116-L124

**Comment by shamil-gadelshin** [comment]
It's perfectly valid to alter those mock implementations for the `MemberOriginValidator` in pallets if you need it. I often didn't use the `membership` pallet because most of the time I didn't test the membership pallet, and mock implementations simplify tests initialization.

**Comment by L3pereira** [comment]
Well I forgot the mock is inside ensure_bounty_actor_manger and there is some logic inside of it, what makes these tests nessessary.
But yes the mock implementation should not interfere. What will interfere with the result of these tests, is what we discussed in #3329

**Comment by bedeho** [comment]
But @shamil-gadelshin  does this mean that all unit tests so far did not actually test for cases where someone tried to sign with the wrong account for the given member?

**Comment by L3pereira** [comment]
Real implementation is dependent on https://github.com/Joystream/joystream/blob/1c37a713511cfaf35b9577069f78827492324f2c/runtime-modules/bounty/src/actors.rs#L35
So you are also dependent on unit tests in Membership pallet being able to test some signing/member existing cases.
I removed the part of the test that steps in this enum branch  https://github.com/Joystream/joystream/blob/1c37a713511cfaf35b9577069f78827492324f2c/runtime-modules/bounty/src/actors.rs#L33-L38
I think doesn't make sense, since I would be testing a mock not the real implementation

Basically (in the bounty module)  you can only test  the case when the actor is council (BadOrigin errors)https://github.com/Joystream/joystream/blob/1c37a713511cfaf35b9577069f78827492324f2c/runtime-modules/bounty/src/actors.rs#L39-L43

**Comment by shamil-gadelshin** [comment]
> But @shamil-gadelshin does this mean that all unit tests so far did not actually test for cases where someone tried to sign with the wrong account for the given member?

There are tests in the `membership` pallet for those use cases. It doesn't make sense to test the membership pallet in all other pallets again - we only test that the mock implementation is called.
